From 9d291e5a9425e23dd90c6fd79081bafd5bcd4c32 Mon Sep 17 00:00:00 2001
From: David Michael <david.michael@coreos.com>
Date: Sun, 7 Jan 2018 11:05:06 -0500
Subject: [PATCH] arm64: Set -fno-PIC along with -mcmodel=large

As seen in GCC's gcc/config/aarch64/aarch64.c, -fPIC with large
code model is unsupported.  This fixes the "sorry, unimplemented"
errors when building with compilers defaulting to -fPIC.
---
 purgatory/arch/arm64/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/purgatory/arch/arm64/Makefile b/purgatory/arch/arm64/Makefile
index 636abea..80068ca 100644
--- a/purgatory/arch/arm64/Makefile
+++ b/purgatory/arch/arm64/Makefile
@@ -1,6 +1,7 @@
 
 arm64_PURGATORY_EXTRA_CFLAGS = \
 	-mcmodel=large \
+	-fno-PIC \
 	-fno-stack-protector \
 	-fno-asynchronous-unwind-tables \
 	-Wundef \
-- 
2.14.3

